### PR TITLE
resolve BUGFIX "Too many instances .." is reported for rpc output

### PR DESCRIPTION
**Pull request by fanchanghu**
Hi,
I met the same issue with #538 , and I'm trying to fix it, can you see if my pull request is adequate for this? 

@michalvasko

### CODE REVIEW

**Comment by fanchanghu** [comment]
In addition, This issue is produced  when there are nodes with same name in rpc input and output, and node in input is wrongly used for building output data tree.

File: src/resolve.c [hunk 1]
```diff
@@ -2659,6 +2659,19 @@ resolve_partial_json_data_nodeid(const char *nodeid, const char *llist_value, st
     goto parse_predicates;
 
     while (1) {
+        /* find the correct parent schema node for rpc */
+        sparent = (start && start->parent) ? start->parent->schema : NULL;
```
Comment (Prasann1506) [inline src/resolve.c]: sparent : This pointer is not declared

Comment (fanchanghu) [reply src/resolve.c]: oh... I forgot it when I copy the code. It should be fixed now.

Comment (fanchanghu) [reply src/resolve.c]: I just checked your issue again, it seems that is's not same with mine, and may not be fixed by this request. Sorry for that...

**Comment by michalvasko** [comment]
Hi,
firstly, I do not think mixing `input` and `output` nodes in one data tree is a good idea. But I agree that it should be handled some way.

Could you maybe provide the steps that you are creating the data tree with? Did this pull request solve your issue?

Regards,
Michal

**Comment by fanchanghu** [comment]
Hi,
the module:
[container-list-leaf.yang.txt](https://github.com/CESNET/libyang/files/2248833/container-list-leaf.yang.txt)

the code:
```
#include <stdlib.h>
#include <stdio.h>
#include <libyang/libyang.h>

#define assert(x) if(!(x)){printf("assert fail: " #x "\n");return 1;}else{printf("assert success: " #x "\n");}

int main()
{
    struct ly_ctx *ctx = NULL;
    const struct lys_module *m = NULL;
    struct lyd_node *req = NULL, *root = NULL, *d = NULL;
    int rc = 0;

    ctx = ly_ctx_new(NULL, LY_CTX_NOYANGLIBRARY);
    m = lys_parse_path(ctx, "container-list-leaf.yang", LYS_IN_YANG);
    assert(m)
    req = lyd_parse_path(ctx, "container-list-leaf.xml", LYD_XML, LYD_OPT_RPC, NULL);
    assert(req)

    root = lyd_new_path(NULL, ctx, "/ListLeaf:setdata", NULL, LYD_ANYDATA_CONSTSTRING, 0);
    assert(root)
    d = lyd_new_path(root, ctx, "/ListLeaf:setdata/interfaces", NULL, LYD_ANYDATA_CONSTSTRING, LYD_PATH_OPT_OUTPUT);
    assert(d)
    d = lyd_new_path(root, ctx, "/ListLeaf:setdata/interfaces/ifEntry[ifDescr='thisisone']", NULL, LYD_ANYDATA_CONSTSTRING, LYD_PATH_OPT_OUTPUT);
    assert(d)
    d = lyd_new_path(root, ctx, "/ListLeaf:setdata/interfaces/ifEntry[ifDescr='thisisone']/ifMtu", (void*)"2", LYD_ANYDATA_CONSTSTRING, LYD_PATH_OPT_OUTPUT);
    assert(d)
    d = lyd_new_path(root, ctx, "/ListLeaf:setdata/interfaces/ifEntry[ifDescr='thisisone']/ifIndex", (void*)"2", LYD_ANYDATA_CONSTSTRING, LYD_PATH_OPT_OUTPUT);
    assert(d)

    rc = lyd_validate(&root, LYD_OPT_RPCREPLY, NULL);
    assert(!rc)

    lyd_free(root);
    ly_ctx_destroy(ctx, NULL);

    printf("finish\n");

    return 0;
}

```
Then libyang will report something like:
```
libyang[0]: Too many instances of "interfaces" in "output". (path: /ListLeaf:setdata/interfaces)
```

**Comment by michalvasko** [comment]
Hi,
I think it should be fixed now.

Regards,
Michal

**Pull request closed without merging**